### DISCUSSION TITLE
does minio support dynamic scale?

### DISCUSSION THREAD

**Discussion by lddlww** [minio/minio#14602] [Q&A] [answered]
when i add new minio node to inital minio cluster, i have to stop initial minio cluster and then start 

how can i scale initial minio cluster online ?

**Comment by klauspost** [comment, marked as answer]
Expansion is done by adding server pools, typically of the size of the current install:

https://docs.min.io/minio/baremetal/installation/expand-minio-distributed.html

**Reply by lddlww** [reply]
however i must restart minio service for scale

**Reply by klauspost** [reply]
Yes, you need that for upgrades as well, and you shouldn't plan to upgrade more often than every 2 years, otherwise your planning is wrong. Shouldn't take more than max a few seconds.

Read https://docs.min.io/minio/baremetal/installation/expand-minio-distributed.html#considerations

**Reply by lddlww** [reply]
thanks，read https://docs.min.io/minio/baremetal/installation/expand-minio-distributed.html#expansion-is-non-disruptive

i am confuse about why restart minio cluster is non-disruptive to applications and ongoing operations？

**Reply by klauspost** [reply]
> why restart minio cluster is non-disruptive to applications and ongoing operations

S3 Clients should be capable of retrying operations. So while some requests may be interrupted the clients should retry.

**Reply by lddlww** [reply]
understood, thank you very much
